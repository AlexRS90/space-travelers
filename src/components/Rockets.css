.container34 {
  display: flex;
  flex-direction: column;
  gap: 30px;
  align-items: center;
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.wrapper {
  display: flex;
  width: 80%;
  padding: 2%;
  border-radius: 10px;
}

.wrapper img {
  border-radius: 10px;
}

.info-wrapper {
  padding-left: 20px;
}

.info-wrapper h2 {
  color: #314e72;
}

.info-wrapper p {
  color: #314e72;
}

.book-rocket-btn {
  border-radius: 10px;
  border: 1px solid teal;
  padding: 10px;
  color: #314e72;
  background-color: #7a9cc6;
}

.visible {
  display: block;
}

.hidden {
  display: none;
}

.badge {
  color: #fff;
  background-color: green;
}