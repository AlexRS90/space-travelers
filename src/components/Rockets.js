import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRocket } from '../redux/rocket/rocket';
import Button from './bookRocket';
import Cancel from './cancelRocket';
import './Rockets.css';
/*eslint-disable*/

const Rockets = () => {
  const dispatch = useDispatch();
  const rocket = useSelector((state) => state.rockets);
  useEffect(() => {
    dispatch(fetchRocket());
  }, []);
  return (
    <div className="container34">
      {rocket.map((cohete) => (
        <div key={cohete.id} className="wrapper">
          <img src={cohete.image} alt={`Rocket: ${cohete.name}`} width="300" />
          <div className="info-wrapper">
            <h2>{cohete.name}</h2>
            <p><span className={(cohete.reserved === true ? 'badge' : 'hidden')}>Reserved</span> {cohete.description}</p>
            <span className={(cohete.reserved === false ? 'visible' : 'hidden')}><Button id={cohete.id} /></span>
            <span className={(cohete.reserved === true ? 'visible' : 'hidden')}><Cancel id={cohete.id} /></span>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Rockets;
