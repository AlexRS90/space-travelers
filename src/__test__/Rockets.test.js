import { render, screen, cleanup } from '@testing-library/react';
// import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets';

afterEach(() => {
  cleanup();
});

describe('render the component to the DOM', () => {
  test('render rockets component', async () => {
    render(<Provider store={store}><Rockets /></Provider>);
    const rocketElement = await screen.findByTestId('testing');
    expect(rocketElement).toBeInTheDocument();
  });
});
