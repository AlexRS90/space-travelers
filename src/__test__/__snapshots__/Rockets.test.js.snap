// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the component to the DOM Match snapshot 1`] = `
<div
  className="container34"
  data-testid="testing"
/>
`;
